Authentication issue when transfering a component

**What happened**:

Transfering a component version returns `unable to add artifact: failed to push: failed to authorize: failed to fetch oauth token: unexpected status: 405 Method Not Allowed
exit status 1`

This is not happening every time, so there seems to be some inconsistency there.

flux output:

```sh
⋊> ~/g/s/g/o/MPAS on bootstrap-component ⨯ FLUX_VERSION=v2.0.0-rc.5 GITHUB_TOKEN=xxx GITHUB_USERNAME=souleb make release-bootstrap-component                        
go run ./cmd/release-bootstrap-component/main.go --flux-version v2.0.0-rc.5 --repository-url ghcr.io/souleb/mpas-bootstrap-component \
        --username souleb
We are going to package the bootstrap component and ship it as an OCM component.

INFO[0001] adding resource blob                          logger=ocm resource=flux
INFO[0001] adding resource blob                          logger=ocm resource=ocm-config
INFO[0001] transferring version                          history= logger=ocm version="\"github.com/souleb/flux:v2.0.0-rc.5\""
transferring version "github.com/souleb/flux:v2.0.0-rc.5"...
INFO[0003]   transferring resources                      history= logger=ocm version="\"github.com/souleb/flux:v2.0.0-rc.5\""
...resource 0...
INFO[0003] adding resource blob                          logger=ocm resource=flux
...resource 1...
INFO[0003] adding resource blob                          logger=ocm resource=ocm-config
...resource 2(fluxcd/source-controller:v1.0.0-rc.5)...
INFO[0004] adding resource blob                          logger=ocm resource=source-controller
Failed to transfer component:  github.com/souleb/flux:v2.0.0-rc.5: transferring resource 2: unable to add blob (component github.com/souleb/flux:v2.0.0-rc.5 resource source-controller): unable to add blob (OCI repository component-descriptors/github.com/souleb/flux): failed to push: unexpected status: 405 Method Not Allowed
exit status 1
make: *** [release-bootstrap-component] Error 1
```

**What you expected to happen**:

 I would expect this to have consistent behaviour, it should keep failing it it fails once.

**How to reproduce it (as minimally and precisely as possible)**:

use https://github.com/open-component-model/MPAS/pull/48 and run
```sh
FLUX_VERSION=v2.0.0-rc.5 GITHUB_TOKEN=xxx GITHUB_USERNAME=xxx make release-bootstrap-component
```

**Anything else we need to know**:

The way we setup credentials is based on https://github.com/open-component-model/ocm/blob/main/examples/lib/cred2/example.go

```go
	consumerID := credentials.NewConsumerIdentity(identity.CONSUMER_TYPE,
		identity.ID_HOSTNAME, regURL.Host,
		identity.ID_PATHPREFIX, c.username,
	)

	creds := credentials.DirectCredentials{
		credentials.ATTR_USERNAME: c.username,
		credentials.ATTR_PASSWORD: c.token,
	}

	c.Context.OCMContext().CredentialsContext().SetCredentialsForConsumer(consumerID, creds)
```

**Environment**:
mac os

It still fails intermittently:

```sh
INFO[0001] adding resource blob                          logger=ocm resource=flux
INFO[0001] adding resource blob                          logger=ocm resource=ocm-config
INFO[0001] transferring version                          history= logger=ocm version="\"github.com/souleb/flux:v2.0.0-rc.5\""
transferring version "github.com/souleb/flux:v2.0.0-rc.5"...
Failed to transfer component:  github.com/souleb/flux:v2.0.0-rc.5: creating target version: failed to authorize: failed to fetch oauth token: unexpected status: 405 Method Not Allowed
exit status 1
make: *** [release-bootstrap-component] Error 1
```

--- a/examples/lib/example.go
+++ b/examples/lib/example.go
@@ -22,7 +22,7 @@ const resourceName = "package"
 func MyFirstOCMApplication() error {
 	octx := ocm.DefaultContext()
 
-	repoSpec := ocireg.NewRepositorySpec("ghcr.io/open-component-model/ocm", nil)
+	repoSpec := ocireg.NewRepositorySpec("ghcr.io/open-component-model/ocm")
 
 	repo, err := octx.RepositoryForSpec(repoSpec)
 	if err != nil {

--- a/pkg/contexts/ocm/repositories/genericocireg/type.go
+++ b/pkg/contexts/ocm/repositories/genericocireg/type.go
@@ -81,10 +81,10 @@ type ComponentRepositoryMeta struct {
 }
 
 func NewComponentRepositoryMeta(subPath string, mapping ComponentNameMapping) *ComponentRepositoryMeta {
-	return &ComponentRepositoryMeta{
+	return DefaultComponentRepositoryMeta(&ComponentRepositoryMeta{
 		ComponentNameMapping: mapping,
 		SubPath:              subPath,
-	}
+	})
 }
 
 ////////////////////////////////////////////////////////////////////////////////

--- a/pkg/contexts/ocm/repositories/ocireg/type.go
+++ b/pkg/contexts/ocm/repositories/ocireg/type.go
@@ -5,8 +5,11 @@
 package ocireg
 
 import (
+	"strings"
+
 	"github.com/open-component-model/ocm/pkg/contexts/oci/repositories/ocireg"
 	"github.com/open-component-model/ocm/pkg/contexts/ocm/repositories/genericocireg"
+	"github.com/open-component-model/ocm/pkg/utils"
 )
 
 // ComponentNameMapping describes the method that is used to map the "Component Name", "Component Version"-tuples
@@ -29,10 +32,20 @@ type ComponentRepositoryMeta = genericocireg.ComponentRepositoryMeta
 type RepositorySpec = genericocireg.RepositorySpec
 
 // NewRepositorySpec creates a new RepositorySpec.
-func NewRepositorySpec(baseURL string, meta *ComponentRepositoryMeta) *RepositorySpec {
+// If no ocm meta is given, the subPath part is extracted from the base URL.
+// Otherwise, the given URL is used as OCI registry URL as it is.
+func NewRepositorySpec(baseURL string, metas ...*ComponentRepositoryMeta) *RepositorySpec {
+	meta := utils.Optional(metas...)
+	if meta == nil {
+		if idx := strings.Index(baseURL, "/"); idx > 0 {
+			meta = NewComponentRepositoryMeta(baseURL[idx+1:])
+			baseURL = baseURL[:idx]
+		}
+	}
+
 	return genericocireg.NewRepositorySpec(ocireg.NewRepositorySpec(baseURL), meta)
 }
 
-func NewComponentRepositoryMeta(subPath string, mapping ComponentNameMapping) *ComponentRepositoryMeta {
-	return genericocireg.NewComponentRepositoryMeta(subPath, mapping)
+func NewComponentRepositoryMeta(subPath string, mapping ...ComponentNameMapping) *ComponentRepositoryMeta {
+	return genericocireg.NewComponentRepositoryMeta(subPath, utils.OptionalDefaulted(OCIRegistryURLPathMapping, mapping...))
 }

